src/cache/assignment-cache-factory.ts

```diff
@@ -0,0 +1,24 @@
+import type { AssignmentCache } from '@eppo/js-client-sdk-common';
+
+import SimpleAssignmentCache from './simple-assignment-cache';
+import HybridAssignmentCache from './hybrid-assignment-cache';
+import { AsyncStorageAssignmentCache } from './async-storage-assignment-cache';
+
+export function assignmentCacheFactory({
+  forceMemoryOnly = false,
+  storageKeySuffix,
+}: {
+  forceMemoryOnly?: boolean;
+  storageKeySuffix: string;
+}): AssignmentCache {
+  const simpleCache = new SimpleAssignmentCache();
+  if (forceMemoryOnly) {
+    return simpleCache;
+  }
+  const asyncStorageCache = new AsyncStorageAssignmentCache(storageKeySuffix);
+  const hybridAssignmentCache = new HybridAssignmentCache(
+    simpleCache,
+    asyncStorageCache
+  );
```

Comment:
In the js-client-sdk, we are using a `simpleCache` (`servingStore`) + `localStorageCache` (`persistentStore`) (https://github.com/Eppo-exp/js-client-sdk/blob/dced1c2ded84a2173e6394542b16dd700da6931e/src/cache/assignment-cache-factory.ts#L31). For react-native, we swap out the `localStorage` implementation for one that uses `@react-native-async-storage/async-storage`:

`simpleCache` (`servingStore`) + `asyncStorageCache ` (`persistentStore`)